### PR TITLE
Finetuned openai azure models cost calculation

### PR DESCRIPTION
**Pull request by nirkopler**
**Description:**
Add cost calculation for fine tuned **Azure** with relevant unit tests.
see https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/fine-tuning?tabs=turbo&pivots=programming-language-studio for more information.
this PR is the result of this PR: https://github.com/langchain-ai/langchain/pull/12190

Twitter handle: @nirkopler

### CODE REVIEW

**Comment by nirkopler** [comment]
@mspronesti This is the fix for finetuned Azure models

File: libs/langchain/langchain/callbacks/openai_info.py
Comment (mspronesti) [inline libs/langchain/langchain/callbacks/openai_info.py:102]: This should work, however allow me to suggest splitting on `".ft"` instead of a single dot, which might be part of the model's name (in the future).

Comment (baskaryan) [reply libs/langchain/langchain/callbacks/openai_info.py:102]: also for safety/clarity could we update the below check in line 103 to `if ":ft-" in model_name`

Comment (nirkopler) [reply libs/langchain/langchain/callbacks/openai_info.py:102]: @baskaryan @mspronesti yes I agree, Ill fix it and push 🙃

**Review by mspronesti — APPROVED** [review]
lgtm

Edit: notice I don't have write permissions here.   So despite the PR looks good to me, you need approval from the code owners.

**Comment by baskaryan** [comment]
thank you @nirkopler @mspronesti!

**Pull request merged**
